app/src/androidTest/java/com/google/jetpackcamera/VideoRecordingDeviceTest.kt

```diff
@@ -70,6 +73,54 @@ internal class VideoRecordingDeviceTest {
             activityScenario!!.result
         }.register("key", TEST_CONTRACT) { result ->
             assert(!result)
+            assert(activityScenario!!.result.resultCode == Activity.RESULT_CANCELED)
+        }.launch(uri)
+    }
+
+    @Test
+    fun video_capture_external_intent() = run {
+        val timeStamp = System.currentTimeMillis()
+        val uri = getTestUri(timeStamp)
+        getTestRegistry {
+            activityScenario = ActivityScenario.launchActivityForResult(it)
+            // Wait for the capture button to be displayed
+            uiDevice.wait(
+                Until.findObject(By.res(CAPTURE_BUTTON)),
+                5000
+            )
+            uiDevice.findObject(By.res(CAPTURE_BUTTON)).longClick()
```

Comment:
How long is a `longClick()`? If we don't have a sufficiently long click, then the recording may fail due to not having enough data. Maybe we should use a different UiAutomator API to perform the long press?